src/main/java/searching_system/SearcherManager.java

```diff
@@ -0,0 +1,35 @@
+package searching_system;
+
+import data_access_storage.*;
+import entities.*;
+import matching_system.*;
+
+import java.util.ArrayList;
+
+public class SearcherManager implements SearcherInputBoundary{
+    private final Searcher searcher;
+
+    final UserRepoManager userRepoManager;
+
+    final SearcherOutputBoundary searcherOutputBoundary;
+
+    public SearcherManager(SearcherOutputBoundary searcherOutputBoundary, SearchRequestModel searchRequestModel, UserRepoManager userRepoManager){
+        this.userRepoManager = userRepoManager;
+        this.searcherOutputBoundary = searcherOutputBoundary;
+        ArrayList<UserAccount> userAccountArrayList = new ArrayList<UserAccount>();
+        for(RequestModel rm: this.userRepoManager.getAllUserAccount().values()){
+            userAccountArrayList.add(rm.getUserAccount());
+        }
+        this.searcher = new Searcher(searchRequestModel.getKeyword(), userAccountArrayList);
+
+    }
+    @Override
```

Comment:
It's a bit weird that SearcherManager takes SearchRequestModel as a parameter.  It's more naturally  that SearcherManager Constructor just initializes instance variables, and move the rest of code to the create method.